examples/arithmetics/src/language-server/arithmetics-util.ts

```diff
@@ -4,9 +4,11 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
-export function applyOp(op: '+' | '-' | '*' | '/'): (x: number, y: number) => number {
+export function applyOp(op: '+' | '-' | '*' | '/' | '^' | '%'): (x: number, y: number) => number {
     if (op === '+') return (x, y) => x + y;
     if (op === '-') return (x, y) => x - y;
     if (op === '*') return (x, y) => x * y;
+    if (op === '^') return (x, y) => Math.pow(x, y);
+    if (op === '%') return (x, y) => x % y;
     return (x, y) => y === 0 ? x : x / y;
```

Comment:
Instead of the `if`s you might use a switch statement, like
```suggestion
    switch(op) {
        case '+':  return (x, y) => x + y;
        case '-':  return (x, y) => x - y;
        case '*':  return (x, y) => x * y;
        case '/':  return (x, y) => x / y;
        case '*':  return (x, y) => x * y;
        case '/':  return (x, y) => x / y;
        case '^':  return (x, y) => Math.pow(x, y);
        case '%':  return (x, y) => x % y;
        case '==': return (x, y) => x == y ? 1 : 0;
        case '!=': return (x, y) => x != y ? 1 : 0;
        case '<':  return (x, y) => x  < y ? 1 : 0;
        case '<=': return (x, y) => x <= y ? 1 : 0;
        case '>':  return (x, y) => x  > y ? 1 : 0;
        case '>=': return (x, y) => x >= y ? 1 : 0;
    }
```

Reply:
Yeah, switch-case statements would be great to simplify this case, I added them!